### PR TITLE
Template Part Block: Use `get_block_file_template` for rendering

### PR DESCRIPTION
**Pull request by ockham**
## What?
Change the Template Part block's render method (`render_block_core_template_part`) to use `get_block_file_template` rather than a number of lower-level functions.

## Why?
Higher-level functions such as [`get_block_file_template`](https://developer.wordpress.org/reference/functions/get_block_file_template/) apply filters (such as the eponymous [`get_block_file_template`](https://developer.wordpress.org/reference/hooks/get_block_file_template/)) that otherwise aren't invoked. (This is relevant e.g. for https://github.com/WordPress/gutenberg/pull/51449.)

Furthermore, it can be argued that if a high-level function is available to what otherwise requires calling a number of lower-level functions, it's preferable to use the high-level function (unless there are side effects and/or performance issues, of course) 😄 

## How?
By using `get_block_file_template` (which returns a `WP_Block_Template` object) in the code branch that loads a block template from a theme file, thus supplanting calls to the lower-level `_get_block_template_file` and `_inject_theme_attribute_in_block_template_content` functions.

Note that [`get_block_file_template` invokes both of them in its function body](https://github.com/WordPress/wordpress-develop/blob/20806d4d7fed387f26deb134085f59e91ee66364/src/wp-includes/block-template-utils.php#L1121-L1181). (`_inject_theme_attribute_in_block_template_content` [is called](https://github.com/WordPress/wordpress-develop/blob/20806d4d7fed387f26deb134085f59e91ee66364/src/wp-includes/block-template-utils.php#L560) via [`_build_block_template_result_from_file`](https://github.com/WordPress/wordpress-develop/blob/20806d4d7fed387f26deb134085f59e91ee66364/src/wp-includes/block-template-utils.php#L541-L585)).

## Testing Instructions
Verify that Template Parts still work as before:
- On the frontend, with an unmodified block theme;
- In the editor -- loading, modifying, and saving;
- On the frontend, with modifications to the template part.

### CODE REVIEW

**Review by felixarntz — APPROVED** [review]
Thanks @ockham!

**Comment by spacedmonkey** [comment]
Please don't merge this. This change does not set the area variable which makes a regression from my last pr

**Comment by spacedmonkey** [comment]
The new function doesn't seem to
Support parent themes. Did you test with a parent theme?

**Comment by ockham** [comment]
> The new function doesn't seem to Support parent themes. Did you test with a parent theme?

I just did, and it seemed to work fine in my testing. (I made a quick child theme of TT3 locally and verified that it picked up the parent's template parts; I then added a modified `header.html` template part to the child and verified that it overrode the parent's.)

I don't quite see why `get_block_file_template` wouldn't support child themes 🤔 -- after all, it's using [`_get_block_template_file`](https://github.com/WordPress/wordpress-develop/blob/d7aff966e60a1a3bd4683d6f9f0754f49cf721d3/src/wp-includes/block-template-utils.php#L244-L298) to locate the file, just like we used to do directly in the block's render method. But please LMK if I'm missing something!

**Comment by ockham** [comment]
Since this PR has approval and [parent themes seem to be working fine for me](https://github.com/WordPress/gutenberg/pull/52892#issuecomment-1648768489), I'll go ahead and merge (in order to unblock https://github.com/WordPress/gutenberg/pull/51449).

Happy to tweak or revert in case we find that it's breaking something after all!

**Comment by johnbillion** [comment]
This introduced a regression: #56013

**Pull request merged**
